treeherder/perf/email.py

```diff
@@ -0,0 +1,222 @@
+        return asdict(self)
+
+
+class EmailWriter(ABC):
+    def __init__(self):
+        self._email = Email()
+
+    def prepare_new_email(self, must_mention: Union[List[object], object]) -> dict:
+        """
+        Template method
+        """
+        must_mention = self.__ensure_its_list(must_mention)
+
+        self._write_address()
+        self._write_subject()
+        self._write_content(must_mention)
+        return self.email
+
+    @property
+    def email(self):
+        return self._email.as_payload()
+
+    @abstractmethod
+    def _write_address(self):
+        pass  # pragma: no cover
+
+    @abstractmethod
+    def _write_subject(self):
+        pass  # pragma: no cover
+
+    @abstractmethod
+    def _write_content(self, must_mention: List[object]):
+        pass  # pragma: no cover
+
+    @staticmethod
+    def __ensure_its_list(must_mention) -> List[object]:
+        if not isinstance(must_mention, List):
+            must_mention = [must_mention]
+        return must_mention
+
+
+# For automatically backfilling performance data
+class BackfillReportContent:
+    DESCRIPTION = """Perfherder automatically backfills performance jobs originating from Linux platforms.
+     It does this every hour, as long as it doesn't exceed the daily limit.
+    > __Here's a summary of latest backfills:__
+    ---
+        """
+
+    TABLE_HEADERS = """
+    | Alert summary | Alert | Job symbol | Total backfills (aprox.) | Push range |
+    | :---: | :---: | :---: | :---: | :---: |
+        """
+
+    def __init__(self):
+        self._raw_content = None
+
+    def include_records(self, records: List[BackfillRecord]):
+        self._initialize_report_intro()
+
+        for record in records:
+            self._include_in_report(record)
+
+    def _initialize_report_intro(self):
+        if self._raw_content is None:
+            self._raw_content = self.DESCRIPTION + self.TABLE_HEADERS
+
+    def _include_in_report(self, record: BackfillRecord):
+        new_table_row = self._build_table_row(record)
+        self._raw_content += f"{new_table_row}\n"
+
+    def _build_table_row(self, record: BackfillRecord) -> str:
+        alert_summary = record.alert.summary
+        alert = record.alert
+        job_type = record.job_type
+        total_backfills = record.total_backfills_triggered
+        push_range = self.__build_push_range(record.get_context())
+
+        # some fields require adjustments
+        summary_id = alert_summary.id
+        alert_id = alert.id
+        job_symbol = str(job_type)
+
+        return f"| {summary_id} | {alert_id} | {job_symbol} | {total_backfills} | {push_range} |"
+
+    def __build_push_range(self, backfill_context: List[dict]) -> str:
+        """
+        Provides link to Treeherder' s Job view
+        """
+        from_push, to_push = self.__fetch_border_pushes(backfill_context)  # suspect range as tuple
+
+        repository = from_push.repository.name
+        from_change = from_push.revision
```

Comment:
We have the revisions directly in the context as `push__revision`, but we still needed to get the repository anyway, so it's not bad to have the entire push object

Reply:
I've added a comment so we can address this in future iterations.